frontend/src/components/dashboard/create/email/EmailTemplate.tsx

```diff
@@ -105,16 +105,24 @@ const EmailTemplate = ({
       <SaveDraftModal
         saveable
         onSave={async () => {
-          if (subject && body && from) {
-            await handleSaveTemplate(true)
+          if (campaignId && subject && body && from) {
+            await saveTemplate(+campaignId, subject, body, replyTo, from)
```

Comment:
One other side effect of `handleSaveTemplate` is that it calls `setErrorMsg`. This is such that when users click "Edit draft" (this is not working at the moment) when they faced an error during the finish later flow, they can see the error message on the template page. Not sure if this behaviour should be preserved. 

![image](https://user-images.githubusercontent.com/3666479/100593981-0f1ff280-3334-11eb-8345-65ddcefeb11d.png)

Reply:
I believe we can retain this behaviour as when `saveTemplate()` fails, the page should not be navigated to the dashboard hence the component should not be dismounted - we can set state on the component without warnings. I have amended it as following, this would throw a warning as of now, but should be resolved with #871 .
https://github.com/opengovsg/postmangovsg/blob/a0a9971cb8e17fb346a13bf7e252b10d841ea9d7/frontend/src/components/dashboard/create/email/EmailTemplate.tsx#L101-L110